src/main/kotlin/uk/gov/justice/digital/hmpps/incentivesapi/service/PrisonerIepLevelReviewService.kt

```diff
@@ -122,6 +122,28 @@ class PrisonerIepLevelReviewService(
     }
   }
 
+  private suspend fun getIepLevelForReviewType(prisonerInfo: PrisonerAtLocation, reviewType: ReviewType): String {
+    val iepLevelsForPrison = iepLevelService.getIepLevelsForPrison(prisonerInfo.agencyId)
+    when (reviewType) {
+      ReviewType.INITIAL -> {
+        return iepLevelsForPrison.first(IepLevel::default).iepLevel
+      }
+
+      ReviewType.TRANSFER -> {
+        val currentIepLevelForPrisoner = getCurrentIEPLevelForPrisoners(bookingIds = listOf(prisonerInfo.bookingId), useNomisData = true, useClientCredentials = true)
```

Comment:
Do we know if at the point of the this `TRANSFER` message being processed the current prisoner IEP level is the one at the old prison (before transfer) or if it's already set to the new prison default somehow?

Don't know if my question makes sense.

Reply:
That is a really good point. I think we need to do some digging into Prison API for a transferred prisoner to see how the incentive level history looks

Reply:
[Writeup of investigation](https://dsdmoj.atlassian.net/browse/INC-672?focusedCommentId=235075)